filebeat/harvester/log.go

```diff
@@ -68,12 +72,28 @@ func (h *Harvester) Harvest() {
 		// Partial lines return error and are only read on completion
 		ts, text, bytesRead, jsonFields, err := readLine(processor)
 		if err != nil {
+
 			if err == reader.ErrFileTruncate {
 				logp.Info("File was truncated. Begin reading file from offset 0: %s", h.Path)
 				h.SetOffset(0)
 				return
 			}
 
+			if err == reader.ErrRemoved {
+				logp.Info("File was removed: %s. Closing because close_removed is enabled.", h.Path)
+				return
+			}
+
+			if err == reader.ErrRenamed {
+				logp.Info("File was renamed: %s. Closing because close_renamed is enabled.", h.Path)
+				return
+			}
+
+			if err == io.EOF {
+				logp.Info("End of file reached: %s. Closing because close_eof is enabled.", h.Path)
```

Comment:
oh, it's ErrInactive. How about using switch statement on error types (as all errors are variables):

```
switch err {
    case reader.ErrFileTruncate:
      ...
    case reader.ErrRemoved:
      logp.Info(...)
    case reader.ErrRenamed:
      logp.Info(...)
    case reader.ErrInactive:
      logp.Info(...)
    case io.EOF:
      logp.Info(...)
    default:
        logp.Info("Read line error: %s", err)
}
return
```